storage/remote/codec.go

```diff
@@ -352,6 +353,21 @@ func LabelProtosToMetric(labelPairs []*prompb.Label) model.Metric {
 	return metric
 }
 
+// uniqueLabelNameProtos creates a []prompb.Label with duplicate label names removed
```

Comment:
We should error in this case, not hide the issue.

Reply:
Good point. I'll add a check for duplicate labels in `validateLabelsAndMetricName` and update the tests I wrote accordingly.